### PR TITLE
[NetApp] Update to 2022-11-01

### PR DESCRIPTION
**Pull request by audunn**
**Related command**
az netappfiles

**Description**
Update NetApp Files (ANF) to api-version 2022-11-01 

**Testing Guide**

**History Notes**

[NetAppFiles] `az netappfiles volume`: Add new command `get-groupid-list-for-ldapuser` to Get Group Id List for LDAP User
[NetAppFiles] `az netappfiles account update`: Add paramter `identity_type`
[NetAppFiles] `az netappfiles volume update`: Add paramter `snapshot-dir-visible`. If enabled (true) the volume will contain a read-only snapshot directory which provides access to each of the volume's snapshots (defaults to true)

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
NetApp

File: src/azure-cli/azure/cli/command_modules/netappfiles/custom.py [hunk 1]
```diff
@@ -476,7 +650,9 @@ def create_snapshot(cmd, client, resource_group_name, account_name, pool_name, v
                     location=None, no_wait=False):
     location = location or _get_location_from_resource_group(cmd.cli_ctx, resource_group_name)
     body = Snapshot(location=location)
-    return sdk_no_wait(no_wait, client.begin_create, resource_group_name, account_name, pool_name, volume_name, snapshot_name, body)
+    poller = client.begin_create(resource_group_name, account_name, pool_name, volume_name, snapshot_name, body)
+    LongRunningOperation(cmd.cli_ctx)(poller)
+    return sdk_no_wait(no_wait, client.get, resource_group_name, account_name, pool_name, volume_name, snapshot_name)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/netappfiles/custom.py:655]: The get operation is not LongRunningOperation, you should not use use sdk_no_wait for it.

File: src/azure-cli/azure/cli/command_modules/netappfiles/custom.py [hunk 2]
```diff
@@ -441,6 +569,52 @@ def add_export_policy_rule(instance, allowed_clients, unix_read_only, unix_read_
...
+    params = VolumePatch(
+        export_policy=volume_export_policy,
+        service_level=volGetResponse.service_level,
+        usage_threshold=volGetResponse.usage_threshold)
+    # _update_mapper(volGetResponse, params, ['export_policy'])
+    poller = client.begin_update(resource_group_name=resource_group_name, account_name=account_name, pool_name=pool_name, volume_name=volume_name,
+                                 body=params)
+    LongRunningOperation(cmd.cli_ctx)(poller)
+    # export_policy=client.get(resource_group_name, account_name, pool_name, volume_name).export_policy
+    return sdk_no_wait(no_wait, client.get, resource_group_name, account_name, pool_name, volume_name)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/netappfiles/custom.py:615]: same concern here

File: src/azure-cli/azure/cli/command_modules/netappfiles/custom.py [hunk 3]
```diff
@@ -192,14 +288,34 @@ def create_pool(cmd, client, account_name, pool_name, resource_group_name, servi
...
+        size = int(size) * tib_scale
+
+    body = CapacityPoolPatch(qos_type=qos_type, size=size, tags=tags, cool_access=cool_access)
+    logger.debug("ANF Log: PATCH  patch_pool_custom pool body")
+    logger.debug(body)
+    poller = client.begin_update(resource_group_name=resource_group_name, account_name=account_name, pool_name=pool_name,
+                                 body=body)
+    LongRunningOperation(cmd.cli_ctx)(poller)
+    logger.debug("ANF Log: GET updated Response: ")
+    return sdk_no_wait(no_wait, client.get, resource_group_name, account_name, pool_name)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/netappfiles/custom.py:316]: same concern here

File: src/azure-cli/azure/cli/command_modules/netappfiles/custom.py (hunk 1 above)
Comment (audunn) [reply src/azure-cli/azure/cli/command_modules/netappfiles/custom.py:655]: Agreed, thought that was wierd.  Will change

**Comment by audunn** [comment]
@kairu-ms updated

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**